starknet_py/contract.py

```diff
@@ -51,24 +51,23 @@ def from_abi(address: int, abi: ABI) -> "ContractData":
 
 @add_sync_methods
 @dataclass(frozen=True)
-class InvocationResult:
+class SentTransaction:
     """
-    Dataclass returned from invocation. Contains basic details and allows waiting for transaction's acceptance.
+    Dataclass exposing the interface of transaction related to a performed action
     """
 
     hash: CastableToHash
-    contract: ContractData
     _client: "Client"
     status: Optional[str] = None
     block_number: Optional[int] = None
 
     async def wait_for_acceptance(
```

Comment:
It has to return same type as self when subclassed. We probably need a `TypeVar` here.
```
...self: T, ...): T
```

Reply:
Unless it works like this already of course :)